homeassistant/components/fibaro/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for Fibaro integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from fiblary3.common.exceptions import HTTPException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import ConfigType
+
+from . import FibaroController
+from .const import CONF_IMPORT_PLUGINS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_IMPORT_PLUGINS, default=False): bool,
+    }
+)
+
+
+def _connect_to_fibaro(data: dict[str, Any]) -> FibaroConnectResult:
+    """Validate the user input allows us to connect to fibaro."""
+    controller = FibaroController(data)
+    connected = controller.connect()
+    if connected:
+        _LOGGER.debug(
+            "Successful connection to fibaro home center with url %s", data[CONF_URL]
+        )
+        return FibaroConnectResult(connected, controller.hub_serial, controller.name)
```

Comment:
Suggest returning a tuple of `connected, controller`.  Then you can skip the entire FibaroConnectResult...unless there is some purpose for it that I am missing.

Reply:
Thanks for spotting that, tuple makes absolutely sense here.